packages/server-wallet/src/utilities/messaging.ts

```diff
@@ -19,43 +19,34 @@ export function mergeOutgoing(outgoing: Notice[]): Notice[] {
 
   const {sender} = outgoing[0].params;
 
-  const mergedOutgoing: Record<string, Payload> = {};
-
-  for (const notice of outgoing) {
-    const {recipient, data} = notice.params as {recipient: string; data: Payload};
-    if (!mergedOutgoing[recipient]) {
-      mergedOutgoing[recipient] = {walletVersion: WALLET_VERSION};
+  const messages = outgoing.map(o => o.params as WireMessage);
+
+  return _.map(
+    _.groupBy(messages, o => o.recipient),
+    (rcptMsgs, recipient) => {
+      const states = uniqueAndSorted(rcptMsgs.flatMap(n => n.data.signedStates || []));
+      const requests = uniqueAndSorted(rcptMsgs.flatMap(n => n.data.requests || []));
+      const objectives = uniqueAndSorted(rcptMsgs.flatMap(n => n.data.objectives || []));
```

Comment:
This plus L38-40 makes me think that `n.data.signedStates` should _always_ be an array (maybe of length 0)?